### ISSUE TITLE
Avoid unhandledRejection on api calls?

### ISSUE THREAD

**Issue by ujja** [googleapis/nodejs-dialogflow-cx#235]
Hello,

for example in detect intent text sample there is unhandledRejection handler. I see that some errors are coming there. For example if I turn off WiFi then I got one:

`unhandledRejection: request to https://europe-west2-dialogflow.googleapis.com/$rpc/google.cloud.dialogflow.cx.v3.Sessions/DetectIntent failed, reason: getaddrinfo ENOTFOUND europe-west2-dialogflow.googleapis.com`

This unhandledRejection confuses our application. Is there a way to handle it locally?

**Comment by meredithslota** [comment]
@ujja Can you share which version of the library you're using?

**Comment by ujja** [comment]
3.5.0

See also
https://github.com/googleapis/gax-nodejs/issues/1302

**Comment by galz10** [comment]
Hi Ujja, i'm a little confused about what `handling this locally` means. I believe you can call the API inside of a try-catch and do something depending on the error for example 

```
try {
  // Call API in here 
} catch (e) {
  if (e instanceof unhandledRejection) {
    // statements to handle unhandledRejection exceptions
  } else {
   // Do something else 
  }
}
```

Let us know if this fixed your issue.

**Comment by ujja** [comment]
We already catching all exceptions:

`
    return this.sessionClient.detectIntent(request, this.detectIntentOpts)
      .then((responses) => {
        const response = responses[0]

        if (response.queryResult.diagnosticInfo) {
          response.queryResult.diagnosticInfo = struct.decode(response.queryResult.diagnosticInfo)
        }
      ...
      }).catch((err) => {
        debug(err)
        throw new Error(`...`)
      })

`

An exception is not caught somewhere in the lib.

We use booth dialogflow, and dialogflowCX clients, and we have problems with booth. So maybe problem is in GAX. 

It can happen that it is coming just in legacy mode. (https://github.com/googleapis/gax-nodejs/issues/1302)

**Comment by galz10** [comment]
Yes from the response from Alex i believe this might be a GAX issue and not the client library.
